lang/rs/bindings/cners/examples/loopback/src/main.rs

```diff
@@ -0,0 +1,221 @@
+                .takes_value(true)
+                .allow_hyphen_values(true) // allow negative values
+                .help("Core affinity for loopback thread. If provided, can help to improve performance"),
+        )
+        .get_matches();
+
+    // start logging
+    env_logger::builder()
+        .try_init()
+        .expect("Failed to initialize event logger");
+
+    log::info!("CNE loopback application. Press ctrl-c to exit");
+
+    // Parse config file.
+    let jsonc_file = matches
+        .value_of("config")
+        .unwrap()
+        .parse::<String>()
+        .expect("Invalid CNE JSON file name");
+
+    // Parse port name.
+    let port_name = matches
+        .value_of("port")
+        .unwrap()
+        .parse::<String>()
+        .expect("Invalid port name");
+
+    let mut core_id = -1;
+    if matches.is_present("affinity") {
+        core_id = matches
+            .value_of("affinity")
+            .unwrap()
+            .parse::<i16>()
+            .expect("Invalid core id");
+    }
+
+    // Get CNE instance.
+    let cne = CneInstance::get_instance();
+
+    // Configure CNE.
+    match cne.configure(&jsonc_file) {
+        Err(e) => {
+            log::error!("Error configuring CNE: {}", e.to_string());
+            return;
+        }
+        Ok(_) => {}
+    }
+
+    // Get CNE port to recv/send packets.
+    let port = cne.get_port(&port_name);
+
+    // Check if port is valid.
+    if let Err(ref e) = port {
+        log::error!("{}", e.to_string());
+        // Cleanup and exit.
+        if let Err(e) = cne.cleanup() {
+            log::error!("{}", e.to_string());
+        }
+        return;
+    }
+    let port = port.unwrap();
+
+    // Create ctrl-c channel to check for ctrl-c event.
+    let ctrl_c_events = ctrl_channel().unwrap();
+
+    // Create channel for periodic logging.
+    let ticks = tick(Duration::from_secs(1));
+
+    // Variable to check if application is running.
+    let running = Arc::new(AtomicBool::new(true));
+    let r = running.clone();
+
+    // Spawn thread for loopback packets.
+    let handle = thread::spawn(move || {
+        // Affinitize thread to core.
+        if core_id >= 0 {
+            core_affinity::set_for_current(core_affinity::CoreId {
+                id: core_id as usize,
+            });
+        }
+
+        let tid = CneInstance::register_thread("loopback").unwrap();
+
+        while r.load(Ordering::SeqCst) {
+            match loopback(&port) {
+                Err(e) => {
+                    log::error!("loopback error: {}", e.to_string());
+                    break;
+                }
+                Ok(_) => {}
+            }
+        }
+
+        CneInstance::unregister_thread(tid).unwrap();
+    });
+
+    // log port stats. Wait for ctrl-c event from user.
+    loop {
+        select! {
+            recv(ticks) -> _ => {
+                if running.load(Ordering::SeqCst) {
+                    let port = cne.get_port(&port_name).unwrap();
```

Comment:
Port name again, just use port number. Plus this is a lookup I am guessing and a lookup may consume cycles.

We also need to support multiple ports in this application, but that can be done after these are address in a different PR.

Reply:
Multiple ports in applicationc an be different PR. I want a simple example as part of library crate. We can have example applications outside library crate